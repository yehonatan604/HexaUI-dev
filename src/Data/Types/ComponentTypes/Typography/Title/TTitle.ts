import { TComponent, ComponentOptions } from "../../TComponent";

export type TTitle = TComponent<"h1" | "h2" | "h3" | "h4" | "h5" | "h6"> & {
    options?: TTitleOptions;
};

export type TTitleOptions = ComponentOptions & {
    align?: string;
    size?: "xs" | "sm" | "md" | "lg" | "xl";
    padding?: string;
}