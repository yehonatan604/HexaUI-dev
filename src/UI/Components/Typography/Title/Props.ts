import { TComponent } from "../../../../Data/Types/TComponent";

export type TitleProps = TComponent<"h1" | "h2" | "h3" | "h4" | "h5" | "h6"> & {
    children: string;
    size?: string;
    color?: string;
};