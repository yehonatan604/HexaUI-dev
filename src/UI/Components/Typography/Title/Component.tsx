import useTheme from "../../../../Core/Hooks/useTheme";
import { FlexTypes } from "../../../../Data/Constants/FlexTypes";
import { Sizes } from "../../../../Data/Constants/Sizes";
import { TTitle } from "../../../../Data/Types/ComponentTypes/Typography/Title/TTitle";
import Flex from "../../Layout/Flex/Flex";

const Title = (props: TTitle) => {
  const { children, options, ...componentProps } = props;

  // Hooks
  const { colors } = useTheme();

  // Options
  const size = options?.size || Sizes.Md;
  const textColor = options?.textVariant || colors.standradLight;
  const bgColor = options?.bgVariant || colors.standrad;
  const padding = options?.padding || "p-5";
  const align = options?.align || FlexTypes.Center;

  // Text Size
  let textSize = "xl";
  if (size === Sizes.Sm) {
    textSize = "2xl";
  } else if (size === Sizes.Md) {
    textSize = "3l";
  } else if (size === Sizes.Lg) {
    textSize = "4xl";
  } else if (size === Sizes.Xl) {
    textSize = "5xl";
  }

  // JSX
  return (
    <Flex
      className={`w-full ${bgColor}`}
      options={{
        justify: align,
      }}
    >
      <h1 className={`${padding} text-${textSize} text-${textColor}`} {...componentProps}>
        {children}
      </h1>
    </Flex>
  );
};

export default Title;
