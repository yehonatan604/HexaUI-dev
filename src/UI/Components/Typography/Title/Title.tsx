import { FlexTypes } from "../../../../Data/Constants/FlexTypes";
import { Sizes } from "../../../../Data/Constants/Sizes";
import { TTitle } from "./types/TTitle";
import Flex from "../../Layout/Flex/Flex";

const Title = (props: TTitle) => {
  const { children, options, ...componentProps } = props;

  // Options
  const size = options?.size || Sizes.Md;
  const textColor = options?.textVariant || "standard-l";
  const bgColor = options?.bgVariant || "standard";
  const padding = options?.padding || "p-5";
  const align = options?.align || FlexTypes.Center;

  // Text Size
  let textSize = "xl";
  if (size === Sizes.Sm) {
    textSize = "2xl";
  } else if (size === Sizes.Md) {
    textSize = "3l";
  } else if (size === Sizes.Lg) {
    textSize = "4xl";
  } else if (size === Sizes.Xl) {
    textSize = "5xl";
  }

  // JSX
  return (
    <Flex
      className={`w-full ${bgColor}`}
      options={{
        justify: align,
      }}
    >
      <h1 className={`${padding} text-${textSize} text-${textColor}`} {...componentProps}>
        {children}
      </h1>
    </Flex>
  );
};

export default Title;
