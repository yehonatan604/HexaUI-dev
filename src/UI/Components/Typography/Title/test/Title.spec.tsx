import { render, screen } from "@testing-library/react";
import Title from "../Title"; // Adjust the import path
import { describe, it, expect } from "vitest";

describe("Title Component", () => {
  it("should render the title text correctly", () => {
    render(<Title options={{}}>Test Title</Title>);

    // Check if the title text is rendered
    expect(screen.getByText("Test Title")).toBeInTheDocument();
  });

  it("should apply custom size, text, and background variants", () => {
    render(
      <Title
        options={{
          size: "lg",
          textVariant: "blue-500",
          bgVariant: "gray-100",
          padding: "p-8",
        }}
      >
        Custom Title
      </Title>
    );

    const titleElement = screen.getByText("Custom Title");

    // Check if the correct size, text color, background color, and padding are applied
    expect(titleElement).toHaveClass("text-4xl", "text-blue-500", "p-8");
    const parentElement = titleElement.closest("div");
    expect(parentElement).toHaveClass("w-full", "gray-100");
  });

  it("should align the title based on the 'align' option", () => {
    render(
      <Title
        options={{
          align: "start",
        }}
      >
        Aligned Title
      </Title>
    );

    const parentElement = screen.getByText("Aligned Title").closest("div");

    // Check if the title is aligned to the start
    expect(parentElement).toHaveClass("justify-start");
  });

  it("should render and match the snapshot", () => {
    const { asFragment } = render(
      <Title options={{ size: "xl", textVariant: "red-500" }}>Snapshot Title</Title>
    );

    // Snapshot the Title structure
    expect(asFragment()).toMatchSnapshot();
  });
});
