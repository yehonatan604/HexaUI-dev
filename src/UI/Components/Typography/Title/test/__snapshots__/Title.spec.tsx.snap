// Vitest Snapshot v1

exports[`Title Component > should render and match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-row bg- text- justify-center items-start flex-nowrap w-full standard"
  >
    <h1
      class="p-5 text-5xl text-red-500"
    >
      Snapshot Title
    </h1>
  </div>
</DocumentFragment>
`;
